Userland/Libraries/LibJS/Runtime/Object.cpp

```diff
@@ -190,6 +190,18 @@ bool Object::create_data_property_or_throw(PropertyName const& property_name, Va
     return success;
 }
 
+// 7.3.6 CreateNonEnumerableDataPropertyOrThrow ( O, P, V ), https://tc39.es/proposal-error-cause/#sec-createnonenumerabledatapropertyorthrow
+bool Object::create_non_enumerable_data_property_or_throw(PropertyName const& property_name, Value value)
+{
+    VERIFY(!value.is_empty());
```

Comment:
Let's also `VERIFY` that the `property_name` is valid.

Reply:
added